[Snyk] Upgrade socket.io from 4.6.2 to 4.7.0

Szotkowski opened the pull request:
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to upgrade socket.io from 4.6.2 to 4.7.0.</h3>

:information_source: Keep your dependencies up-to-date. This makes it easier to fix existing vulnerabilities and to more quickly identify and fix newly disclosed vulnerabilities when they affect your project.
<hr/>

- The recommended version is **1 version** ahead of your current version.
- The recommended version was released **23 days ago**, on 2023-06-22.


<details>
<summary><b>Release notes</b></summary>
<br/>
  <details>
    <summary>Package name: <b>socket.io</b></summary>
    <ul>
      <li>
        <b>4.7.0</b> - <a href="https://snyk.io/redirect/github/socketio/socket.io/releases/tag/4.7.0">2023-06-22</a></br><h3>Bug Fixes</h3>
<ul>
<li>remove the Partial modifier from the socket.data type (<a href="https://snyk.io/redirect/github/socketio/socket.io/issues/4740" data-hovercard-type="pull_request" data-hovercard-url="/socketio/socket.io/pull/4740/hovercard">#4740</a>) (<a href="https://snyk.io/redirect/github/socketio/socket.io/commit/e5c62cad60fc7d16fbb024fd9be1d1880f4e6f5f">e5c62ca</a>)</li>
</ul>
<h3>Features</h3>
<h4>Support for WebTransport</h4>
<p>The Socket.IO server can now use WebTransport as the underlying transport.</p>
<p>WebTransport is a web API that uses the HTTP/3 protocol as a bidirectional transport. It's intended for two-way communications between a web client and an HTTP/3 server.</p>
<p>References:</p>
<ul>
<li><a href="https://w3c.github.io/webtransport/" rel="nofollow">https://w3c.github.io/webtransport/</a></li>
<li><a href="https://developer.mozilla.org/en-US/docs/Web/API/WebTransport" rel="nofollow">https://developer.mozilla.org/en-US/docs/Web/API/WebTransport</a></li>
<li><a href="https://developer.chrome.com/articles/webtransport/" rel="nofollow">https://developer.chrome.com/articles/webtransport/</a></li>
</ul>
<p>Until WebTransport support lands <a href="https://snyk.io/redirect/github/nodejs/node/issues/38478" data-hovercard-type="issue" data-hovercard-url="/nodejs/node/issues/38478/hovercard">in Node.js</a>, you can use the <code>@ fails-components/webtransport</code> package:</p>
<div class="highlight highlight-source-js notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="import { readFileSync } from &quot;fs&quot;;
import { createServer } from &quot;https&quot;;
import { Server } from &quot;socket.io&quot;;
import { Http3Server } from &quot;@ fails-components/webtransport&quot;;

// WARNING: the total length of the validity period MUST NOT exceed two weeks (https://w3c.github.io/webtransport/#custom-certificate-requirements)
const cert = readFileSync(&quot;/path/to/my/cert.pem&quot;);
const key = readFileSync(&quot;/path/to/my/key.pem&quot;);

const httpsServer = createServer({
  key,
  cert
});

httpsServer.listen(3000);

const io = new Server(httpsServer, {
  transports: [&quot;polling&quot;, &quot;websocket&quot;, &quot;webtransport&quot;] // WebTransport is not enabled by default
});

const h3Server = new Http3Server({
  port: 3000,
  host: &quot;0.0.0.0&quot;,
  secret: &quot;changeit&quot;,
  cert,
  privKey: key,
});

(async () =&gt; {
  const stream = await h3Server.sessionStream(&quot;/socket.io/&quot;);
  const sessionReader = stream.getReader();

  while (true) {
    const { done, value } = await sessionReader.read();
    if (done) {
      break;
    }
    io.engine.onWebTransportSession(value);
  }
})();

h3Server.startServer();"><pre><span class="pl-k">import</span> <span class="pl-kos">{</span> <span class="pl-s1">readFileSync</span> <span class="pl-kos">}</span> <span class="pl-k">from</span> <span class="pl-s">"fs"</span><span class="pl-kos">;</span>
<span class="pl-k">import</span> <span class="pl-kos">{</span> <span class="pl-s1">createServer</span> <span class="pl-kos">}</span> <span class="pl-k">from</span> <span class="pl-s">"https"</span><span class="pl-kos">;</span>
<span class="pl-k">import</span> <span class="pl-kos">{</span> <span class="pl-v">Server</span> <span class="pl-kos">}</span> <span class="pl-k">from</span> <span class="pl-s">"socket.io"</span><span class="pl-kos">;</span>
<span class="pl-k">import</span> <span class="pl-kos">{</span> <span class="pl-v">Http3Server</span> <span class="pl-kos">}</span> <span class="pl-k">from</span> <span class="pl-s">"@ fails-components/webtransport"</span><span class="pl-kos">;</span>

<span class="pl-c">// WARNING: the total length of the validity period MUST NOT exceed two weeks (https://w3c.github.io/webtransport/#custom-certificate-requirements)</span>
<span class="pl-k">const</span> <span class="pl-s1">cert</span> <span class="pl-c1">=</span> <span class="pl-en">readFileSync</span><span class="pl-kos">(</span><span class="pl-s">"/path/to/my/cert.pem"</span><span class="pl-kos">)</span><span class="pl-kos">;</span>
<span class="pl-k">const</span> <span class="pl-s1">key</span> <span class="pl-c1">=</span> <span class="pl-en">readFileSync</span><span class="pl-kos">(</span><span class="pl-s">"/path/to/my/key.pem"</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-k">const</span> <span class="pl-s1">httpsServer</span> <span class="pl-c1">=</span> <span class="pl-en">createServer</span><span class="pl-kos">(</span><span class="pl-kos">{</span>
  key<span class="pl-kos">,</span>
  cert
<span class="pl-kos">}</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-s1">httpsServer</span><span class="pl-kos">.</span><span class="pl-en">listen</span><span class="pl-kos">(</span><span class="pl-c1">3000</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-k">const</span> <span class="pl-s1">io</span> <span class="pl-c1">=</span> <span class="pl-k">new</span> <span class="pl-v">Server</span><span class="pl-kos">(</span><span class="pl-s1">httpsServer</span><span class="pl-kos">,</span> <span class="pl-kos">{</span>
  <span class="pl-c1">transports</span>: <span class="pl-kos">[</span><span class="pl-s">"polling"</span><span class="pl-kos">,</span> <span class="pl-s">"websocket"</span><span class="pl-kos">,</span> <span class="pl-s">"webtransport"</span><span class="pl-kos">]</span> <span class="pl-c">// WebTransport is not enabled by default</span>
<span class="pl-kos">}</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-k">const</span> <span class="pl-s1">h3Server</span> <span class="pl-c1">=</span> <span class="pl-k">new</span> <span class="pl-v">Http3Server</span><span class="pl-kos">(</span><span class="pl-kos">{</span>
  <span class="pl-c1">port</span>: <span class="pl-c1">3000</span><span class="pl-kos">,</span>
  <span class="pl-c1">host</span>: <span class="pl-s">"0.0.0.0"</span><span class="pl-kos">,</span>
  <span class="pl-c1">secret</span>: <span class="pl-s">"changeit"</span><span class="pl-kos">,</span>
  cert<span class="pl-kos">,</span>
  <span class="pl-c1">privKey</span>: <span class="pl-s1">key</span><span class="pl-kos">,</span>
<span class="pl-kos">}</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-kos">(</span><span class="pl-k">async</span> <span class="pl-kos">(</span><span class="pl-kos">)</span> <span class="pl-c1">=&gt;</span> <span class="pl-kos">{</span>
  <span class="pl-k">const</span> <span class="pl-s1">stream</span> <span class="pl-c1">=</span> <span class="pl-k">await</span> <span class="pl-s1">h3Server</span><span class="pl-kos">.</span><span class="pl-en">sessionStream</span><span class="pl-kos">(</span><span class="pl-s">"/socket.io/"</span><span class="pl-kos">)</span><span class="pl-kos">;</span>
  <span class="pl-k">const</span> <span class="pl-s1">sessionReader</span> <span class="pl-c1">=</span> <span class="pl-s1">stream</span><span class="pl-kos">.</span><span class="pl-en">getReader</span><span class="pl-kos">(</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

  <span class="pl-k">while</span> <span class="pl-kos">(</span><span class="pl-c1">true</span><span class="pl-kos">)</span> <span class="pl-kos">{</span>
    <span class="pl-k">const</span> <span class="pl-kos">{</span> done<span class="pl-kos">,</span> value <span class="pl-kos">}</span> <span class="pl-c1">=</span> <span class="pl-k">await</span> <span class="pl-s1">sessionReader</span><span class="pl-kos">.</span><span class="pl-en">read</span><span class="pl-kos">(</span><span class="pl-kos">)</span><span class="pl-kos">;</span>
    <span class="pl-k">if</span> <span class="pl-kos">(</span><span class="pl-s1">done</span><span class="pl-kos">)</span> <span class="pl-kos">{</span>
      <span class="pl-k">break</span><span class="pl-kos">;</span>
    <span class="pl-kos">}</span>
    <span class="pl-s1">io</span><span class="pl-kos">.</span><span class="pl-c1">engine</span><span class="pl-kos">.</span><span class="pl-en">onWebTransportSession</span><span class="pl-kos">(</span><span class="pl-s1">value</span><span class="pl-kos">)</span><span class="pl-kos">;</span>
  <span class="pl-kos">}</span>
<span class="pl-kos">}</span><span class="pl-kos">)</span><span class="pl-kos">(</span><span class="pl-kos">)</span><span class="pl-kos">;</span>

<span class="pl-s1">h3Server</span><span class="pl-kos">.</span><span class="pl-en">startServer</span><span class="pl-kos">(</span><span class="pl-kos">)</span><span class="pl-kos">;</span></pre></div>
<p>Added in <a href="https://snyk.io/redirect/github/socketio/engine.io/commit/123b68c04f9e971f59b526e0f967a488ee6b0116">123b68c</a>.</p>
<h4>Client bundles with CORS headers</h4>
<p>The bundles will now have the right <code>Access-Control-Allow-xxx</code> headers.</p>
<p>Added in <a href="https://snyk.io/redirect/github/socketio/socket.io/commit/63f181cc12cbbbf94ed40eef52d60f36a1214fbe">63f181c</a>.</p>
<h4>Links</h4>
<ul>
<li>Diff: <a class="commit-link" href="https://snyk.io/redirect/github/socketio/socket.io/compare/4.6.2...4.7.0"><tt>4.6.2...4.7.0</tt></a></li>
<li>Client release: <a href="https://snyk.io/redirect/github/socketio/socket.io-client/releases/tag/4.7.0">4.7.0</a></li>
<li><a href="https://snyk.io/redirect/github/socketio/engine.io/releases/tag/6.5.0"><code>engine.io@~6.5.0</code></a> (<a href="https://snyk.io/redirect/github/socketio/engine.io/compare/6.4.2...6.5.0">diff</a>)</li>
<li><a href="https://snyk.io/redirect/github/websockets/ws/releases/tag/8.11.0"><code>ws@~8.11.0</code></a> (no change)</li>
</ul>
      </li>
      <li>
        <b>4.6.2</b> - <a href="https://snyk.io/redirect/github/socketio/socket.io/releases/tag/4.6.2">2023-05-31</a></br><h3>Bug Fixes</h3>
<ul>
<li><strong>exports:</strong> move <code>types</code> condition to the top (<a href="https://snyk.io/redirect/github/socketio/socket.io/issues/4698" data-hovercard-type="pull_request" data-hovercard-url="/socketio/socket.io/pull/4698/hovercard">#4698</a>) (<a href="https://snyk.io/redirect/github/socketio/socket.io/commit/3d44aae381af38349fdb808d510d9f47a0c2507e">3d44aae</a>)</li>
</ul>
<h4>Links</h4>
<ul>
<li>Diff: <a class="commit-link" href="https://snyk.io/redirect/github/socketio/socket.io/compare/4.6.1...4.6.2"><tt>4.6.1...4.6.2</tt></a></li>
<li>Client release: <a href="https://snyk.io/redirect/github/socketio/socket.io-client/releases/tag/4.6.2">4.6.2</a></li>
<li><a href="https://snyk.io/redirect/github/socketio/engine.io/releases/tag/6.4.2"><code>engine.io@~6.4.2</code></a> (<a href="https://snyk.io/redirect/github/socketio/engine.io/compare/6.4.1...6.4.2">diff</a>)</li>
<li><a href="https://snyk.io/redirect/github/websockets/ws/releases/tag/8.11.0"><code>ws@~8.11.0</code></a> (no change)</li>
</ul>
      </li>
    </ul>
    from <a href="https://snyk.io/redirect/github/socketio/socket.io/releases">socket.io GitHub release notes</a>
  </details>
</details>


<details>
  <summary><b>Commit messages</b></summary>
  </br>
  <details>
    <summary>Package name: <b>socket.io</b></summary>
    <ul>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/00d8ee5b057a8aef830d88da77579037759e762c">00d8ee5</a> chore(release): 4.7.0</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/2dd5fa9dd466c6b39b0764d53fbda6185b4916ce">2dd5fa9</a> ci: add Node.js 20 in the test matrix</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/a5dff0ac83ac5e2b0e518c6d054b2c9b252b2558">a5dff0a</a> docs(examples): increase httpd ProxyTimeout value (2)</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/3035c25982b5bff020cff43fba31bb0cc4cc617c">3035c25</a> docs(examples): increase httpd ProxyTimeout value</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/63f181cc12cbbbf94ed40eef52d60f36a1214fbe">63f181c</a> feat: serve client bundles with CORS headers</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/a250e283daa283c2ed7ffaba3033db717a8142ab">a250e28</a> chore: bump engine.io to version 6.5.0</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/e5c62cad60fc7d16fbb024fd9be1d1880f4e6f5f">e5c62ca</a> fix: remove the Partial modifier from the socket.data type (#4740)</li>
      <li><a href="https://snyk.io/redirect/github/socketio/socket.io/commit/01d37624a8fec5041e0f19e071914f610946a3af">01d3762</a> docs(changelog): update the version range of the engine.io dependency</li>
    </ul>

   <a href="https://snyk.io/redirect/github/socketio/socket.io/compare/faf914c9ab3e06a6e84fc91774a4182e58f8ae70...00d8ee5b057a8aef830d88da77579037759e762c">Compare</a>
  </details>
</details>
<hr/>

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open upgrade PRs.*

For more information:  <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiI2ZDliYzgzMi1mYTVhLTQ4MjAtYjQ3Zi01ODgxYmRjNDIyMTQiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjZkOWJjODMyLWZhNWEtNDgyMC1iNDdmLTU4ODFiZGM0MjIxNCJ9fQ==" width="0" height="0"/>

🧐 [View latest project report](https://app.snyk.io/org/szotkowski/project/36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🛠 [Adjust upgrade PR settings](https://app.snyk.io/org/szotkowski/project/36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d/settings/integration?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🔕 [Ignore this dependency or unsubscribe from future upgrade PRs](https://app.snyk.io/org/szotkowski/project/36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d/settings/integration?pkg&#x3D;socket.io&amp;utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr#auto-dep-upgrades)

<!--- (snyk:metadata:{"prId":"6d9bc832-fa5a-4820-b47f-5881bdc42214","prPublicId":"6d9bc832-fa5a-4820-b47f-5881bdc42214","dependencies":[{"name":"socket.io","from":"4.6.2","to":"4.7.0"}],"packageManager":"npm","type":"auto","projectUrl":"https://app.snyk.io/org/szotkowski/project/36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d?utm_source=github&utm_medium=referral&page=upgrade-pr","projectPublicId":"36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d","env":"prod","prType":"upgrade","vulns":[],"issuesToFix":[],"upgrade":[],"upgradeInfo":{"versionsDiff":1,"publishedDate":"2023-06-22T09:28:44.823Z"},"templateVariants":[],"hasFixes":false,"isMajorUpgrade":false,"isBreakingChange":false,"priorityScoreList":[]}) --->

vendasoft commented on src/keys.ts:
```diff
@@ -5,7 +5,7 @@ import {Credentials} from './repositories/user.repository';
 
 export namespace TokenServiceConstants {
-  export const TOKEN_SECRET_VALUE = '138asda8213';
+  export const TOKEN_SECRET_VALUE = 'up9f3dpq98';
```
toto by melo byt v .env value .. spravne.

Szotkowski replied:
okey

vendasoft commented on src/models/photo.model.ts:
```diff
@@ -0,0 +1,33 @@
+import {Entity, model, property} from '@loopback/repository';
+
+@model()
+export class Photo extends Entity {
```
opravdu potrebujeme model pro fotku? jestlize je tam jen jedna potrebna property -> link ..?

snyk-bot committed:
fix: upgrade socket.io from 4.6.2 to 4.7.0

Snyk has created this PR to upgrade socket.io from 4.6.2 to 4.7.0.

See this package in npm:
https://www.npmjs.com/package/socket.io

See this project in Snyk:
https://app.snyk.io/org/szotkowski/project/36f26f5b-e4bf-4cb8-ae5b-087fbc45bf4d?utm_source=github&utm_medium=referral&page=upgrade-pr
